mypy/fastparse2.py

```diff
@@ -346,23 +349,23 @@ def set_type_optional(self, type: Type, initializer: Expression) -> None:
         if isinstance(type, UnboundType):
             type.optional = optional
 
-    def transform_args(self, n: ast27.arguments, line: int) -> List[Argument]:
+    def transform_args(self, n: ast27.arguments, line: int) -> Tuple[List[Argument], List[Statement]]:
         # TODO: remove the cast once https://github.com/python/typeshed/pull/522
         # is accepted and synced
         type_comments = cast(List[str], n.type_comments)  # type: ignore
         converter = TypeConverter(line=line)
 
-        def convert_arg(arg: ast27.expr) -> Var:
+        decompose_stmts = []  # type: List[Statement]
+        def convert_arg(index: int, arg: ast27.expr) -> Var:
             if isinstance(arg, ast27.Name):
                 v = arg.id
             elif isinstance(arg, ast27.Tuple):
-                # TODO: An `arg` object may be a Tuple instead of just an identifier in the
-                # case of Python 2 function definitions/lambdas that use the tuple unpacking
-                # syntax. The `typed_ast.conversions` module ended up just simply passing the
-                # the arg object unmodified (instead of converting it into more args, etc).
-                # This isn't typesafe, since we will no longer be always passing in a string
-                # to `Var`, but we'll do the same here for consistency.
-                v = arg  # type: ignore
+                v = '__tuple_arg_{}'.format(index + 1)
+                rvalue = NameExpr(v)
+                rvalue.set_line(line)
```

Comment:
Maybe `set_line(arg.lineno, arg.col_offset)`? Then you won't even need the `line` argument to `transform_arg()`.